c/validation/adbc_validation.cc

```diff
@@ -911,6 +911,167 @@ void ConnectionTest::TestMetadataGetObjectsPrimaryKey() {
+      parent_constraint->constraint_column_names[0].data,
+      parent_constraint->constraint_column_names[0].size_bytes);
+  ASSERT_EQ(parent_constraint_column_name, "id");
+
+  // test child table
+  struct AdbcGetObjectsTable* child_table = AdbcGetObjectsDataGetTableByName(
+      *get_objects_data, quirks()->catalog().c_str(), quirks()->db_schema().c_str(),
+      "adbc_fkey_child_test");
+  ASSERT_NE(child_table, nullptr) << "could not find adbc_fkey_child_test table";
+
+  ASSERT_EQ(child_table->n_table_columns, 1);
+  struct AdbcGetObjectsColumn* child_column = AdbcGetObjectsDataGetColumnByName(
+      *get_objects_data, quirks()->catalog().c_str(), quirks()->db_schema().c_str(),
+      "adbc_fkey_child_test", "parent_id");
+  ASSERT_NE(child_column, nullptr)
+      << "could not find parent_id column on adbc_fkey_child_test table";
+
+  ASSERT_EQ(child_table->n_table_constraints, 1)
+      << "expected 1 constraint on adbc_fkey_child_test table, found: "
+      << child_table->n_table_constraints;
+
+  struct AdbcGetObjectsConstraint* child_constraint = child_table->table_constraints[0];
+
+  std::string_view child_constraint_type(child_constraint->constraint_type.data,
+                                         child_constraint->constraint_type.size_bytes);
+  ASSERT_EQ(child_constraint_type, "FOREIGN KEY");
+  ASSERT_EQ(child_constraint->n_column_names, 1)
+      << "expected constraint adbc_fkey_child_test_fkey to be applied to 1 column, "
+         "found: "
+      << child_constraint->n_column_names;
+
+  std::string_view child_constraint_column_name(
+      child_constraint->constraint_column_names[0].data,
+      child_constraint->constraint_column_names[0].size_bytes);
+  ASSERT_EQ(child_constraint_column_name, "parent_id");
+
+  std::string_view child_constraint_column_usage_fk_catalog(
+      child_constraint->constraint_column_usages[0]->fk_catalog.data,
+      child_constraint->constraint_column_usages[0]->fk_catalog.size_bytes);
+  std::string_view child_catalog = child_constraint_column_usage_fk_catalog;
+  std::string_view sqlite_default_catalog = "main";
+  std::string_view postgresql_default_catalog = "postgres";
+
+  ASSERT_THAT(child_catalog, ::testing::AnyOf(::testing::Eq(sqlite_default_catalog),
+                                              ::testing::Eq(postgresql_default_catalog)));
+
+  std::string_view child_constraint_column_usage_fk_db_schema(
+      child_constraint->constraint_column_usages[0]->fk_db_schema.data,
+      child_constraint->constraint_column_usages[0]->fk_db_schema.size_bytes);
+  std::string_view child_db_schema = child_constraint_column_usage_fk_db_schema;
+  std::string_view sqlite_default_db_schema = "";
+  std::string_view postgresql_default_db_schema = "public";
+  ASSERT_THAT(child_db_schema,
+              ::testing::AnyOf(::testing::Eq(sqlite_default_db_schema),
+                               ::testing::Eq(postgresql_default_db_schema)));
```

Comment:
This is either slightly iffy to me, both the current implementation of the test, as well as if it is indeed testing something in the implementation.